system-x/services/filesystem/src/main/java/software/tnb/filesystem/resource/openshift/OpenShiftFileSystem.java

```diff
@@ -0,0 +1,90 @@
+package software.tnb.filesystem.resource.openshift;
+
+import software.tnb.common.config.OpenshiftConfiguration;
+import software.tnb.common.deployment.OpenshiftDeployable;
+import software.tnb.common.openshift.OpenshiftClient;
+import software.tnb.filesystem.service.FileSystem;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import com.google.auto.service.AutoService;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import cz.xtf.core.openshift.OpenShiftWaiters;
+
+@AutoService(FileSystem.class)
+public class OpenShiftFileSystem extends FileSystem implements OpenshiftDeployable {
+    private static final String NAMESPACE = OpenshiftClient.get().getNamespace();
+    private String podLabelValue;
+
+    @Override
+    public void setAppName(String app) {
+        this.podLabelValue = app;
+    }
+
+    @Override
+    public String getFileContent(Path path) {
+        final String podLabelKey = OpenshiftConfiguration.openshiftDeploymentLabel();
+        podIsReady(podLabelKey, podLabelValue);
+        final String podName = getPodName(podLabelKey, podLabelValue);
+        try (InputStream is = OpenshiftClient.get().pods().inNamespace(NAMESPACE).withName(podName).file(path.toString()).read()) {
```

Comment:
add `.withName(podName).inContainer(OpenshiftClient.get().getIntegrationContainer(pod))`, otherwise it might fail for some camel-k integrations that are using knative